test/e2e/dataplane/tcp_pod_to_service.go

```diff
@@ -1,47 +1,77 @@
+		connectorUUID = string(uuid.NewUUID())
 
-	listenerUUID := string(uuid.NewUUID())
-	connectorUUID := string(uuid.NewUUID())
+		By("Creating a listener pod in cluster B, which will wait for a handshake over TCP")
+		listenerPod = f.CreateTCPCheckListenerPod(framework.ClusterB, listenerUUID)
 
-	By("Creating a listener pod in cluster B, which will wait for a handshake over TCP")
-	listenerPod := f.CreateTCPCheckListenerPod(framework.ClusterB, listenerUUID)
+		By("Pointing a service ClusterIP to the listener pod in cluster B")
+		service := f.CreateTCPService(framework.ClusterB, listenerPod.Labels[framework.TestAppLabel], framework.TestPort)
+		framework.Logf("Service for listener pod has ClusterIP: %v", service.Spec.ClusterIP)
 
-	By("Pointing a service ClusterIP to the listener pod in cluster B")
-	service := f.CreateTCPService(framework.ClusterB, listenerPod.Labels[framework.TestAppLabel], framework.TestPort)
-	framework.Logf("Service for listener pod has ClusterIP: %v", service.Spec.ClusterIP)
+		By("Creating a connector pod in cluster A, which will attempt the specific UUID handshake over TCP")
+		connectorPod = f.CreateTCPCheckConnectorPod(framework.ClusterA, listenerPod, service.Spec.ClusterIP, connectorUUID)
 
-	By("Creating a connector pod in cluster A, which will attempt the specific UUID handshake over TCP")
-	connectorPod := f.CreateTCPCheckConnectorPod(framework.ClusterA, listenerPod, service.Spec.ClusterIP, connectorUUID)
+		By("Waiting for the listener pod to exit with code 0, returning what listener sent")
+		exitStatusL, exitMessageL = f.WaitForPodFinishStatus(listenerPod, framework.ClusterB)
+		framework.Logf("Listener output:\n%s", keepLines(exitMessageL, 3))
+		Expect(exitStatusL).To(Equal(int32(0)))
 
-	By("Waiting for the listener pod to exit with code 0, returning what listener sent")
-	exitStatusL, exitMessageL := f.WaitForPodFinishStatus(listenerPod, framework.ClusterB)
-	framework.Logf("Listener output:\n%s", exitMessageL)
-	Expect(exitStatusL).To(Equal(int32(0)))
+		By("Waiting for the connector pod to exit with code 0, returning what connector sent")
+		exitStatusC, exitMessageC = f.WaitForPodFinishStatus(connectorPod, framework.ClusterA)
+		framework.Logf("Connector output\n%s", keepLines(exitMessageC, 2))
+		Expect(exitStatusC).To(Equal(int32(0)))
+	})
+
+	It("Should be able to perform a Pod to Service TCP connection and exchange data between different clusters", func() {
+		By("Verifying what the pods sent to each other contain the right UUIDs")
+		Expect(exitMessageL).To(ContainSubstring(connectorUUID))
+		Expect(exitMessageC).To(ContainSubstring(listenerUUID))
+	})
 
-	By("Waiting for the connector pod to exit with code 0, returning what connector sent")
-	exitStatusC, exitMessageC := f.WaitForPodFinishStatus(connectorPod, framework.ClusterA)
-	framework.Logf("Connector output\n%s", exitMessageC)
-	Expect(exitStatusC).To(Equal(int32(0)))
+	It("Should preserve the source IP", func() {
```

Comment:
LGTM.

I know that we are currently not using POD to POD connectivity in our use-cases, but shall we also have a validation check if pod-to-pod connectivity is working and if the source-ip is preserved. It will help us in identifying if somethings break. What do you say?

Reply:
This is pod to service, yes... 

I'm working first on controlling the scheduling of pods to test combinations of GW + no GW host GW + GW, no GW + no GW ...

then we can also handle that.

Reply:
Cool, thanks.